app/policies/comment_policy.rb

```diff
@@ -80,6 +80,10 @@ def permitted_attributes_for_moderator_create
 
   private
 
+  def user_trusted?
+    Authorizer.for(user: user).accesses_mod_response_templates?
+  end
+
```

Comment:
I'm a little confused - why not just `user.trusted?` here? Or rather as I understand the implementation of [`accesses_mod_response_templates?`](https://github.com/forem/forem/blob/5ef0f436365339bc8b4ab17641e21b1ab38c7118/app/policies/authorizer.rb#L83) (`has_trusted_role? || any_admin? || super_moderator? || tag_moderator?`) it already covers the other branches of the conditional up in `moderator_create?` here

Reply:
It's a good question! There's maybe-sort-of answers here, but I'm **not** particularly convinced this needs to stay — I can see how `trusted?` would simplify this.

* The most basic answer is: I'm [bringing over a similar check in `response_template_policy`](https://github.com/forem/forem/blob/915b3eaca4e70e77943facd3fd8b27f4ae241fe5/app/policies/response_template_policy.rb#L67)
* More significantly: around a year ago, I don't think we _really_ **got** how `trusted users` were _supposed_ to work. (See also: https://github.com/forem/forem/pull/18601) I was brand-new here when I was working on `trusted users -> response templates`, but it wasn't _just_ me: there [was a _lot_ of confusion](https://github.com/forem/forem/pull/17867#discussion_r893461089) about the role at the time.)
* My personal stylistic preference (_especially_ when roles are undergoing a lot of transformation) is to be _very explicit_ about the kinds of permissions being sought/granted, even if they are ultimately derived from a role. This tends to work well with `pundit` (but I think we could do more). (That is: I prefer having views/helpers asking specific questions, like `can_i_do_this_thing?` than scattering `has_role(x) || has_role(y) && not_property(z)` everywhere. When permissions shift, like "trusted users can now access templates", it's easier to change specific questions' answers or introduce new hair-splitting questions than hunt for every `has_role` in the system.)
* Another team (at the time, we had functional teams inside Eng) was working on re-organizing scattered "permission" methods into a (relatively) new class `Authorizer`, so [I added the new method there](https://github.com/forem/forem/blob/915b3eaca4e70e77943facd3fd8b27f4ae241fe5/app/policies/authorizer.rb#L82-L84) — it was awkward, but for better and worse, it felt like a kind-of hybrid between a `Pundit Policy` class and the `rolify` relationship gem. My understanding, at the time, was that we were actively refactoring and reorganizing where permissions (and roles) code lived in the system as well as simultaneously changing some of the roles & permissions.
* We also have had a tendency to aggressively cache role queries, for better and worse, (and the `trusted` role _in particular_ had buggy caching until [rather recently](https://github.com/forem/forem/pull/19275)) — which leads to methods [that equally aggressively route _around_ caching](https://github.com/forem/forem/blob/915b3eaca4e70e77943facd3fd8b27f4ae241fe5/app/policies/authorizer.rb#L86-L101) — being :new: at the time, it wasn't _totally_ clear to me what "sources of truth" we could "trust".
* Due to roles & permissions being in so much flux, it was specifically [unclear at the time](https://github.com/forem/forem/pull/17867#issuecomment-1153595043) whether we could reliably assume "all `admins` are _always_ also `trusted`" (etc.) or if that was only temporarily/accidentally true.

Reply:
Oh! I think I wasn't very clear - what I meant was more that it seems like you could replace the implementation of `moderator_create?` (`user_trusted? || user_moderator? || user_any_admin?`) with just `Authorizer.for(user: user).accesses_mod_response_templates?` since it checks for all of those things